src/analyze/analyze-condition.c

```diff
@@ -72,18 +73,35 @@ static int log_helper(void *userdata, int level, int error, const char *file, in
         return r;
 }
 
-int verify_conditions(char **lines, UnitFileScope scope) {
+int verify_conditions(char **lines, UnitFileScope scope, const char *root) {
         _cleanup_(manager_freep) Manager *m = NULL;
+        _cleanup_free_ char *var = NULL;
+        _cleanup_free_ char **filenames = NULL;
         Unit *u;
         char **line;
         int r, q = 1;
 
+        STRV_FOREACH(line, lines)
+                if (unit_name_is_valid(*line, UNIT_NAME_ANY)) {
+                        r = strv_push(&filenames, *line);
+                        if (r < 0)
+                                return -ENOMEM;
+                }
```

Comment:
hmm, this doesn't look right. i.e. valid condition expressions can very well look exactly like unit file names, i.e. they are not always clearly discernable, afaics

Reply:
We now check first if the string starts with a valid condition type

Reply:
the verb is called "condition", I think it's expected hence that the primary parameter passed then is a condition string. (and the unit name is just a flag)